cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -231,7 +271,7 @@ reader::impl::impl(std::size_t chunk_read_limit,
   // Don't need to do it if we read the file all at once.
   if (_chunk_read_limit > 0) {
     for (auto const& buff : _output_buffers) {
-      _output_buffers_template.emplace_back(column_buffer::empty_like(buff));
+      _output_buffers_template.emplace_back(inline_column_buffer::empty_like(buff));
```

Comment:
Is there a difference between `inline_column_buffer` and `gather_column_buffer` in this case?

Reply:
Somewhere in all the refactoring we lost the delete of the copy constructor and the definition of the move constructor. Should those go back in?  @vuule

Reply:
Oh yeah, that one's on me :grimacing: 
I think it would be enough to delete the copy constructor/assignment in the base class.
Why do we need to define the move constructor for these classes?

Reply:
> Parquet only uses inline_column_buffer

A stupid question: in parquet, string data can be decoded via `string_index_pair` (or the `gather_column_buffer`), right?

Reply:
> Why do we need to define the move constructor for these classes?

STL throws fits if copy is deleted and move isn't defined, it seems.  But it seems happy if both are in the base class...I had tried putting both in the derived classes.